src/targets/gpu/fuse_mlir.cpp

```diff
@@ -301,14 +301,87 @@ struct find_mlir_op
     }
 };
 
+struct find_mlir_standalone_convolution_op
+{
+    auto matcher() const { return match::name("convolution"); }
+
+    void apply(module_pass_manager& mpm, const match::matcher_result& r) const
+    {
+        auto conv_based_op = r.result;
+        // enable only for fp32/fp16/i8 types
+        if(std::any_of(conv_based_op->inputs().begin(), conv_based_op->inputs().end(), [&](auto i) {
+               return not contains(
+                   {shape::type_t::float_type, shape::type_t::half_type, shape::type_t::int8_type},
+                   i->get_shape().type());
+           }))
+            return;
+
+        static size_t counter = 0;
+        module_ref mm         = mpm.create_module("mlir_" + std::to_string(counter++));
+        mm->set_bypass();
+        auto [anchor_op, top_inputs] = fuse_input_ops_and_gemm_based_op(mm, conv_based_op);
+        mm->add_return({anchor_op});
+        mpm.get_module().replace_instruction(
+            conv_based_op, mlir_op{conv_based_op->get_operator()}, top_inputs, {mm});
+    }
+};
+
+MIGRAPHX_DECLARE_ENV_VAR(MIGRAPHX_MLIR_ENABLE_OPS);
+bool is_self_decide() { return env(MIGRAPHX_MLIR_ENABLE_OPS::value()).empty(); }
```

Comment:
This should use `string_value_of`, otherwise this will read the env everytime which is slow.

Also, what does `is_self_decided` mean?

Reply:
Hi,

`MIGRAPHX_MLIR_ENABLE_OPS` means that the user specifies concrete operations which rocMLIR has to handle. For example, MIGRAPHX_MLIR_ENABLE_OPS="conv" will only generate convolutions with rocMLIR; fusing will be switched off.

if `MIGRAPHX_MLIR_ENABLE_OPS` is not specified by a user then we decide **ourselves** (named it as **self_decide**) which operations for rocMLIR to enable - i.e., always use fused operations and, if the underlying GPU is from Navi3x, use standalone convolutions

If you want I can rename `MIGRAPHX_MLIR_ENABLE_OPS` to `MIGRAPHX_MLIR_USE_SPECIFIC_OPS`

Reply:
Sure, either name is find. Can you add a comment to source code to explain how this is supposed to work and what the valid values are? I would expected to use `MIGRAPHX_MLIR_ENABLE_OPS=convolution` not `MIGRAPHX_MLIR_ENABLE_OPS=conv` since thats the way we spell the operator, but if its spelled differently for this variable, we should have some comments to document it.

Reply:
Lets not introduce a another indirection here..
and stick to migraphx op names ?